vlttng/venv.py

```diff
@@ -730,6 +745,25 @@ def _build_lttng_ust(self):
             self._runner.cp_rv(os.path.join(log4j_name, log4j_jar),
                                self._paths.log4j_jar)
 
+        if '--enable-java-agent-all' in project.configure or \
+           '--enable-java-agent-log4j2' in project.configure:
+            log4j2_version = '2.22.1'
+            log4j2_name  = 'log4j-{}'.format(log4j2_version)
+            log4j2_zipfile = 'apache-{}-bin.zip'.format(log4j2_name)
+            log4j2_jars = [
+                'log4j-api-{}.jar'.format(log4j2_version),
+                'log4j-core-{}.jar'.format(log4j2_version),
+                'log4j-1.2-api-{}.jar'.format(log4j2_version),
+            ]
+            _pinfo('Download Apache {}'.format(log4j2_name))
+            self._runner.cd(self._paths.src)
+            self._runner.wget('https://dlcdn.apache.org/logging/log4j/{}/{}'.format(
+                log4j2_version, log4j2_zipfile), log4j2_zipfile)
+            self._runner.mkdir_p(log4j2_name)
+            self._runner.unzip(log4j2_zipfile, log4j2_name)
+            for src_jar, dest_jar in zip(log4j2_jars, self._paths.log4j2_jars):
```

Comment:
NL above.

Also this zip looks very flimsy. You change the order in `log4j2_jars` or `self._paths.log4j2_jars` and this doesn't work anymore. Can you think about some way to infer one from the other?

Reply:
Done.